### ISSUE TITLE
After convert from Toolkit to muxc:TabView lots of Layout cycle detected

### ISSUE THREAD

**Issue by ChrisSchaller** [microsoft/microsoft-ui-xaml#3417]
**Describe the bug**

Converted an LOB application from using the Toolkit [TabView](https://docs.microsoft.com/en-us/windows/communitytoolkit/controls/tabview) over to the new TabView in this project, after doing so many (not all) pages with TabView control now experience UnhandledException: A cycle ocurred while layout out the GUI.

> Layout cycle detected. Layout could not complete.

![image](https://user-images.githubusercontent.com/7519539/95820901-ade89380-0d74-11eb-9789-29f396695212.png)

In this state the app either hangs or crashes, there is no recovery.

**Steps to reproduce the bug**
Converted existing LOB application that was using the TabView from the UWP Toolkit.
Haven't been able to create a minimal representation of this issue yet.

**Expected behavior**
Should be no functional difference between the two TabView controls, if there is any significant breaking change, other than some xaml syntax, I would appreciate the heads up in the form of a guidance doc on how to transition across now that it has been removed from the Toolkit project.

**Version Info**
NuGet package version: 
[Microsoft.UI.Xaml 2.4.3]
still present in 
[Microsoft.UI.Xaml 2.5.0-prerelease.200923002]

| Windows 10 version                  | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (xxxxx)              | Yes   |
| May 2020 Update (19041)            | Yes  |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | Yes |

**Additional context**
It doesn't happen _every_ time, but is most consistent when a TabViewItem has a nested TabView or when chaning back to a TabViewItem that has a GridView child element

**Comment by StephenLPeters** [comment]
Unfortunately this isn't a known issue and with the current repro steps this is going to be very difficult for us to investigate. Is it possible to get more information or a repro app for the team to use?

**Comment by ChrisSchaller** [comment]
In trying to create a minimum reproducible code sample I have identified that the root cause of my issue is to do with an event handler registered on `ScrollViewer.ScrollableHeightProperty` on a `ScrollViewer` that was a child element of an affected tab page.
> The behavior was trying to inject a right margin on the `ScrollViewer` when the scrollbar was visible so that the scrollbar did not show over the top of content.

Although this was not an observable issue with the toolkit version of the application, it is an anti-pattern that is discouraged so I have removed the event handler.

I'll close this issue.
